lfbb_cpp/inc/lfbb_impl.hpp

```diff
@@ -60,18 +65,28 @@ T *LfBb<T, size>::WriteAcquire(const size_t free_required) {
     const size_t linear_space = size - r;
     const size_t linear_free = std::min(free, linear_space);
 
-    /* Try to find enough linear space until the end of the buffer */
+    #ifdef LFBB_TRACE
+    LFBB_TRACE(LFBB_FORMATTER("{} (free_required={}) linear_free {} free_required <= linear_free {}\n", __func__, free_required, linear_free, (free_required <= linear_free)));
```

Comment:
Using the `std::format` interface would heavily limit the utility of this. The `printf()` interface is far more ubiquitous.

Also just having to define `LFBB_TRACE` is much easier.

Reply:
`printf` is a [Turing-complete](https://github.com/carlini/printf-tac-toe) virtual machine doing parsing of format string in runtime, while `std::format` and `fmt::format` are compile-time computations, so they should be more friendly to embedded devices. Also using `std::format` spares me of the mess of debugging variadic templates.

Reply:
`std::format` requires C++20, and returns a `std::string`, a dynamically allocated object. Neither of these are embedded or systems friendly in a lot of cases, and the API is nowhere to be found in third party formatting libraries while the `printf()` API is ubiquitous. 

Additionally the `fmt` and `std` libraries use exceptions for formatting error handling while these may also not be available in embedded systems depending on the configuration.

My idea isn't to reimplement `printf()`, something like
```c
define LFBB_TRACE printf
```
would be enough.